### PR TITLE
Add Geometry and Custom SRID

### PR DESCRIPTION
**Pull request by stevegoddard**
Added the ability to specify additional $geomtype and $srid parameters to geom/geog types. Both parameters have default values to ensure backward compatibbility.

Additional tests added.

### CODE REVIEW

**Comment by stevegoddard** [comment]
I've had a another bash at getting the geom / geog to work with cnstants and also pass the PHPunit tests. I can't get it to work. 

How vital is this to the PR?

Steve

**Comment by phaza** [comment]
It's more nice to have, as it eliminates the possibility to write either geometry or geography wrong.

What is important is for the models to also support geometries, as currently they only support geographies. If we allow geometries to be created in migrations, but don't support them in models, I think the issues list will fill up rather quickly.

So basically we need a way to track what fields/columns are geographies, and which are geometries,

At least I think we have to.

**Comment by stevegoddard** [comment]
Thanks Peter.

I'll take a look at models when I have a little more time. I can't get my head round using a choice of 2 constants in the function call - it has me baffled - so I'll leave that for the mo. If you can give me some pointers I'll happily have another look at it.

Steve

**Comment by antonyharfield** [comment]
Hi guys,

This looks really good! Do you think you will be close to merging this change soon?

Thanks,
Antony

File: src/Schema/Grammars/PostgisGrammar.php [hunk 1]
```diff
@@ -14,7 +17,12 @@ class PostgisGrammar extends PostgresGrammar
      */
     public function typePoint(Fluent $column)
     {
-        return 'GEOGRAPHY(POINT, 4326)';
+        // return 'GEOGRAPHY(POINT, 4326)'; 
+        if ((in_array(strtoupper($column->geomtype), $this->allowed_geom_types)) && (is_int((int) $column->srid))) {
+            return strtoupper($column->geomtype) . '(POINT, ' . $column->srid . ')';
+        } else {
+            return dd('Error with validation of geom type or srid! (If geom type is GEOGRAPHY then the SRID must be 4326)');
```
Comment (jonnybarnes) [inline src/Schema/Grammars/PostgisGrammar.php]: Looking through this, surely all these `dd`s should be `Exception`s instead?

**Comment by v1r0x** [comment]
Any news on this? We'd love to have geometry support for our project. Do you need any help/testing/etc.? Would love to help to get this merged! 😄

**Comment by v1r0x** [comment]
Ping! We need geometry support in one of our projects and we like this Plugin, so we don't want to switch to another laravel-based solution (if there's any). Is there anything I can do to help? Since this PR is based on an older version of Laravel, I could create a new PR with support for 5.4 and 5.3 (the one we currently use). I could also have a look at the models (see @phaza comment). Any interest in this?

**Comment by phaza** [comment]
This PR only covers half the problem, there's no code here which makes geometry work in models.
If you need this @v1r0x, you should probably fork this PR and make it complete.

**Comment by v1r0x** [comment]
Thanks for your fast reply. I'm new to Laravel plugin development, could you point me in the right direction @phaza ?

**Comment by v1r0x** [comment]
Regarding the model stuff

**Comment by njbarrett** [comment]
This PR is superseded by https://github.com/njbarrett/laravel-postgis/pull/76

**Pull request closed without merging**